### DISCUSSION TITLE
[knitr] Quick reporting: Build a report based on an R script | /knitr/demo/stitch/

### DISCUSSION THREAD

**Discussion by yihui** [rbind/yihui#769] [Comments]
https://yihui.org/knitr/demo/stitch/

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
Thanks! I'm aware of that. I have fixed it locally a while ago but have not pushed it to the server. Should be fixed now.

> Originally posted on 2012-06-22 05:00:23

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
Just like using PHP to connect to MySQL, you can use R to connect to PostgresSQL (take a look at the database related packages like RODBC and RPostgreSQL). Then the rest of things are similar: just like write HTML with PHP, you write HTML with R. You can use either R Markdown or R HTML as the authoring format.

> Originally posted on 2012-12-01 23:12:27

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
When the data has changed, just re-compile the R Markdown document. I'm not sure what else you were expecting.

> Originally posted on 2012-12-02 04:24:12

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
Yes, you can do it in command line: Rscript -e "library(knitr); knit('your_report.Rnw')"

RStudio is only one possible editor for R Markdown files. It calls R in the above way behind the scene.

> Originally posted on 2012-12-03 07:07:54

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
Google for "cron job" (if you use *nix) or Windows Task Scheduler (if you are under Windows). Or use Shell scripts or Makefile; there are tons of ways to do it if you are familiar with command line tools.

Replace Rnw with Rmd; that is just a typo.

> Originally posted on 2012-12-03 17:37:55

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
yes, use the chunk option results='asis'

> Originally posted on 2013-10-27 05:54:54

**Comment by Tarious14** [comment]
Hello Yihui, you said that "need to write documentclass after #'" when converting to other formats and I don't really get it. Could you give an example for this ? Thanks a lot.

**Reply by yihui** [reply]
I mean when you want to convert an R script to `Rnw`, you may need to declare the LaTeX document class in the beginning, e.g.,

```r
#' \documentclass{article}
```

**Reply by Tarious14** [reply]
Oh I used to test it without backslash like latex command. Almost get it :vv. Thank you for your help and also your terrific package.

**Reply by Tarious14** [reply]
Hi Yihui, it's doesn't work with me. I declared the Latex document class in the first line of my R script and it returns this error:
```
Error in report && format %in% c("Rnw", "Rtex") && !grepl("^\\s*\\\\documentclass"
```
How can I overcome it ? Thanks

**Reply by yihui** [reply]
This is a bug in **knitr**, and I just fixed it: https://github.com/yihui/knitr/commit/ec4c9b632e1b35994340848884a31da2ca24640a Thanks for the report!

```r
remotes::install_github('yihui/knitr')
```
